app/models/location.rb

```diff
@@ -10,6 +10,7 @@ class Location < ApplicationRecord
   validates :location_url, presence: true
 
   def convert_location_url
+    return unless self.name
     self.location_url = self.name.downcase.parameterize
```

Comment:
I think I would use a regex like `\A[a-z\d][a-z\d-]*[a-z\d]\z/i` to remove the bad characters.